main/FatesInventoryInitMod.F90

```diff
@@ -940,4 +955,92 @@ subroutine set_inventory_edcohort_type1(csite,bc_in,css_file_unit,npatches, &
       return
    end subroutine set_inventory_edcohort_type1
 
+   ! ====================================================================================
+
+   subroutine write_inventory_type1(currentSite)
+
+       ! --------------------------------------------------------------------------------
+       ! This subroutine writes the cohort/patch inventory type files in the "type 1"
+       ! format.  Note that for compatibility with ED2, we chose an old type that has
+       ! both extra unused fields and is missing fields from FATES. THis is not
+       ! a recommended file type for restarting a run.
+       ! The files will have a lat/long tag added to their name, and will be
+       ! generated in the run folder.
+       ! --------------------------------------------------------------------------------
+
+       use shr_file_mod, only        : shr_file_getUnit
+       use shr_file_mod, only        : shr_file_freeUnit
+       
+       ! Arguments
+       type(ed_site_type),intent(in), target :: currentSite
+       
+       ! Locals
+       type(ed_patch_type), pointer          :: currentpatch
+       type(ed_cohort_type), pointer         :: currentcohort
+       
+       character(len=128)                    :: pss_name_out         ! output file string
+       character(len=128)                    :: css_name_out         ! output file string
+       integer                               :: pss_file_out
+       integer                               :: css_file_out
+       integer                               :: ilat_int,ilat_dec    ! for output string parsing
+       integer                               :: ilon_int,ilon_dec    ! for output string parsing
+       character(len=32)                     :: patch_str
+       character(len=32)                     :: cohort_str
+       integer                               :: ipatch
+       integer                               :: icohort
+       
+       ! Generate pss/css file name based on the location of the site
+       ilat_int = floor(currentSite%lat)
+       ilat_dec = int(100000*(currentSite%lat - real(ilat_int,r8)))
+       ilon_int = floor(currentSite%lon)
+       ilon_dec = int(100000*(currentSite%lon - real(ilon_int,r8)))
+       
+       write(pss_name_out,'(A8,i3.3,A,i5.5,A2,i3.3,A,i5.5,A5)') &
+             'pss_out_',ilat_int,'.',ilat_dec,'N_',ilon_int,'.',ilon_dec,'E.txt'
+       write(css_name_out,'(A8,i3.3,A,i5.5,A2,i3.3,A,i5.5,A5)') &
+             'css_out_',ilat_int,'.',ilat_dec,'N_',ilon_int,'.',ilon_dec,'E.txt'
```

Comment:
@rgknox is `currenSite` lat and lon range from 0 to 360 or -180 to 180?

Reply:
hmmm, that is a good question.  I suppose -100 and greater will exceed the string size on ilon_int.  I can bump that up to 4.4

Reply:
Also won't `floor` for a negative number result in an erroneous output?  E.g. -63.59 will return ilat_int = -64.  I think we want `int` here.

Reply:
Well, we want to just preserve the number before the decimal.  I'm thinking it will now take the following combination:

ilat_int = floor(abs(currentSite%lat))

This gives it back it's correct sign:
sign( ilat_int, int(currentSite%lat) )

Reply:
Won't `int()` retain the sign as well?  I was thinking something like:
```
ilat_int = int(currentSite%lat)
ilat_dec = abs(int(100000*(currentSite%lat - real(ilat_int,r8))))
```
I thing we need to add `abs()` to the `_dec` too.

Reply:
actually, sorry, your right.  INT() looks like it is a floor towards zero.  I was thinking of NINT(), which rounds up sometimes.